packages/amplify-cli-npm/binary.ts

```diff
@@ -101,37 +101,37 @@ export class Binary {
   public binaryPath: string;
   public installDirectory: string;
   constructor() {
-    this.installDirectory = join(os.homedir(), '.amplify', 'bin');
+    this.installDirectory = path.join(os.homedir(), '.amplify', 'bin');
 
-    if (!existsSync(this.installDirectory)) {
-      mkdirSync(this.installDirectory, { recursive: true });
+    if (!fs.existsSync(this.installDirectory)) {
+      fs.mkdirSync(this.installDirectory, { recursive: true });
     }
 
     let amplifyExecutableName = 'amplify';
-    if (getPlatformBinaryName().endsWith('.exe')) {
+    if (os.type() === 'Windows_NT') {
       amplifyExecutableName = 'amplify.exe';
     }
```

Comment:
nit: I'd sacrifice readability to make `amplifyExecutableName` a `const`, but either way is fine.

```suggestion
    const amplifyExecutableName = os.type() === 'Windows_NT' ? 'amplify.exe' : 'amplify';
```